x-pack/plugins/rollup/server/lib/jobs_compatibility.js

```diff
@@ -0,0 +1,98 @@
+/*
+* Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+* or more contributor license agreements. Licensed under the Elastic License;
+* you may not use this file except in compliance with the Elastic License.
+*/
+
+import { isEqual } from 'lodash';
+
+/**
+ * Checks if given job configs are compatible by attempting to merge them
+ *
+ * @param jobs
+ * @returns {boolean}
+ */
+export function areJobsCompatible(jobs = []) {
+  if(!jobs || !Array.isArray(jobs)) return false;
+  if(jobs.length <= 1) return true;
+
+  try {
+    mergeJobConfigurations(jobs);
+  } catch(e) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Attempts to merge job configurations into a new configuration object keyed
+ * by aggregation, then by field
+ *
+ * @param jobs
+ * @returns {{}}
+ */
+export function mergeJobConfigurations(jobs = []) {
+  if(!jobs || !Array.isArray(jobs) || !jobs.length) {
+    throw new Error('No capabilities available');
+  }
+
+  const allAggs = {};
+
+  // For each job, look through all of its fields
+  jobs.forEach(job => {
+    const fields = job.fields;
+    const fieldNames = Object.keys(fields);
+
+    // Check each field
+    fieldNames.forEach(fieldName => {
+      const fieldAggs = fields[fieldName];
+
+      // Look through each field's capabilities (aggregations)
+      fieldAggs.forEach(agg => {
+        const aggName = agg.agg;
+
+        // If we currently don't have this aggregation, add it.
+        // Special case for date histogram, since there can only be one
+        // date histogram field.
+        if(
+          !allAggs[aggName] ||
+          (!allAggs[aggName][fieldName] && aggName !== 'date_histogram')
```

Comment:
How about extracting these into variables to make the code more self-descriptive?

```js
        const aggDoesntExist = !allAggs[aggName];
        const fieldDoesntExist = !allAggs[aggName][fieldName];
        const aggIsntDateHistogram = aggName !== 'date_histogram';
        if(
          aggDoesntExist ||
          (fieldDoesntExist && aggIsntDateHistogram)
        ) {
          allAggs[aggName] = allAggs[aggName] || {};
          allAggs[aggName][fieldName] = { ...agg };
        }
```

Reply:
if `allAggs[aggName]` is undefined, `const fieldDoesntExist = !allAggs[aggName][fieldName];` will throw an error. will add the other vars tho!